Tests/Algorithm/AlgorithmRegisterIndicatorTests.cs

```diff
@@ -0,0 +1,157 @@
+                        Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                        expected++;
+                    }
+                    else if (key == typeof(TradeBar))
+                    {
+                        var indicator = (indicatorTest as CommonIndicatorTests<TradeBar>).Indicator;
+                        Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                        expected++;
+                    }
+                    var actual = _algorithm.SubscriptionManager.Subscriptions.FirstOrDefault().Consolidators.Count;
+                    Assert.AreEqual(expected, actual);
+                }
+            }
+        }
+
+        [Test, Ignore]
+        public void RegistersIndicatorProperlyPython()
+        {
+            var expected = 0;
+            foreach (var group in _indicatorTestsGroups)
+            {
+                var key = group.Key;
+                foreach (var type in group)
+                {
+                    var indicatorTest = Activator.CreateInstance(type);
+                    if (key == typeof(IndicatorDataPoint))
+                    {
+                        using (Py.GIL())
+                        {
+                            var indicator = (indicatorTest as CommonIndicatorTests<IndicatorDataPoint>).Indicator.ToPython();
+                            Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                        }
+                        expected++;
+                    }
+                    else if (key == typeof(IBaseDataBar))
+                    {
+                        using (Py.GIL())
+                        {
+                            var indicator = (indicatorTest as CommonIndicatorTests<IBaseDataBar>).Indicator.ToPython();
+                            Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                        }
+                        expected++;
+                    }
+                    else if (key == typeof(TradeBar))
+                    {
+                        using (Py.GIL())
+                        {
+                            var indicator = (indicatorTest as CommonIndicatorTests<TradeBar>).Indicator.ToPython();
+                            Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                        }
+                        expected++;
+                    }
+                    var actual = _algorithm.SubscriptionManager.Subscriptions.FirstOrDefault().Consolidators.Count;
+                    Assert.AreEqual(expected, actual);
+                }
+            }
+        }
+
+        [Test, Ignore]
+        public void RegisterPythonCustomIndicatorProperly()
+        {
+            using (Py.GIL())
+            {
+                var module = PythonEngine.ModuleFromString("x",
```

Comment:
Does `PythonEngine` keep track of all modules names? If we reload another module named `x`, what will happen? Just thinking it may be wise to use more descriptive names here.

Reply:
Totally agree. From now on, I think we should will use `Guid.NewGuid().ToString()`.